File: MyClub/MyClub.Model/Responses/DashboardCountResponse.cs
using System;

namespace MyClub.Model.Responses
{
    public class DashboardCountResponse
    {
        public int TotalCount { get; set; }
        public int ThisMonth { get; set; }
        public int LastMonth { get; set; }
        public decimal PercentageChange { get; set; }
    }
} 

File: MyClub/UI/myclub_desktop/lib/models/dashboard/dashboard_count_response.dart
class DashboardCountResponse {
  final int totalCount;
  final int thisMonth;
  final int lastMonth;
  final double percentageChange;

  DashboardCountResponse({
    required this.totalCount,
    required this.thisMonth,
    required this.lastMonth,
    required this.percentageChange,
  });

  factory DashboardCountResponse.fromJson(Map<String, dynamic> json) {
    return DashboardCountResponse(
      totalCount: json['totalCount'] ?? 0,
      thisMonth: json['thisMonth'] ?? 0,
      lastMonth: json['lastMonth'] ?? 0,
      percentageChange: json['percentageChange']?.toDouble() ?? 0.0,
    );
  }
}
